client/src/components/ui/QRScanner.svelte

```diff
@@ -0,0 +1,110 @@
+        await qrScanner?.start();
+    }
+
+    function stopCamera() {
+        camStart = false;
+        return qrScanner?.stop();
+    }
+
+    async function handleFileInput() {
+        const uploadElement = document.getElementById('file-selector');
+        assert(uploadElement instanceof HTMLInputElement);
+        assert(uploadElement.files instanceof FileList);
+        const [first] = uploadElement.files;
+        assert(typeof first !== 'undefined');
+        try {
+            const scanData = await QrScanner.scanImage(first, { returnDetailedScanResult: true });
+            if (scanData.data === '') return;
+            maybeId = DocumentSchema.shape.id.parse(scanData.data.trim());
+            dispatch(Events.OnDocumentScan, maybeId);
+        } catch (err) {
+            topToastMessage.enqueue({ title: 'Failed to Find QR code', body: JSON.stringify(err) });
+        }
+        uploadElement.value = '';
+    }
+
+    function setup() {
+        const videoElement = document.getElementById('scan');
+        assert(videoElement instanceof HTMLVideoElement);
+
+        return new QrScanner(
+            videoElement,
+            result => {
+                if (result.data === '') return;
+                maybeId = DocumentSchema.shape.id.parse(result.data.trim());
+                stopCamera();
+                dispatch(Events.OnDocumentScan, maybeId);
+            }, {
+                highlightCodeOutline: true,
+                highlightScanRegion: true,
+                onDecodeError: (err) => {
+                    if (err === 'No QR code found') return;
+                    topToastMessage.enqueue({ title: 'Failed to Find QR code', body: JSON.stringify(err) });
+                },
+                maxScansPerSecond: 1,
+                returnDetailedScanResult: true,
+            }
+        );
+    }
+    
+    onMount(() => {
+        // Needs to be here so that the <video> tag is rendered before it is asserted
+        qrScanner = setup();
+    });
+
+    onDestroy(() => {
+        // Otherwise, when the window is closed - browser still thinks you are recording.
+        stopCamera();
+    });
+</script>
+
+<video id='scan'><track kind="captions"></video>
+{#await QrScanner.hasCamera()}
+    Loading cameras.
+    {:then hasCamera}
+        {#if !hasCamera}
+            No cameras detected.
+        {:else}
+            <header>
+                {#if camStart}
+                    <Button type={ButtonType.Secondary} on:click={() => stopCamera()}> End Capture </Button>
+                {:else}
+                    <Button on:click={() => startCamera()}> Start Capture </Button>
+                {/if}
+            </header>
+        {/if}
+{/await}
+<header>
+    Upload a file with the QR code: <input on:change={() => handleFileInput()} type="file" id="file-selector" accept="image/*">
+</header>
+<section>
+    {#if maybeId === null || maybeId === ''} 
+        No valid QR code detected
+    {/if}
+</section>
```

Comment:
Let's move the entire `<section>` inside the conditional. On that note, is `<section>` really the most appropriate element that we can use to just display text?
```suggestion
{#if maybeId === null || maybeId === ''}
    <section>
        No valid QR code detected
    </section>
{/if}
```